lms/services/blackboard_api/service.py

```diff
@@ -53,12 +56,34 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        for file_ in files:
+            file_["id"] = f"blackboard://content-resource/{file_['id']}/"
+
+        return files
+
+    def public_url(self, course_id, file_id):
+        # Extract just the {file_id} part out of the
+        # "blackboard://content-resource/{file_id}/" string.
+        file_id = file_id.split("/")[3]
```

Comment:
`file_id.split("/")[3]` is a little unsatisfactory but we do know for sure that this is a valid URL in that particular format since we created it, so I think it's actually fine. A regex might be more readable though.

Reply:
Actually I think the code is nicer if we _never_ parse `blackboard://...` URLs. We're given a `blackboard://...` URL for the file that we're looking for. We can wrap all the file IDs from the Blackboard API into `blackboard://...` URLs and then compare _those_ to the `blackboard://...` URL that we were given. That way we only ever need to format `blackboard://...` URLs, we never need to parse them.